.github/workflows/update-coverage.yml

```diff
@@ -0,0 +1,47 @@
+name: Update coverage
+
+on:
+  schedule:
+    - cron: 0 0 * * *
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  run-tests:
+    name: Run tests
+    uses: ./.github/workflows/run-tests.yml
+
+  update-coverage:
+    if: ${{ needs.run-tests.outputs.active-branch-coverage > needs.run-tests.outputs.default-branch-coverage }}
+    name: Update coverage
+    runs-on: ubuntu-latest
+    needs:
+      - run-tests
+    env:
+      ACTIVE_BRANCH_COVERAGE: ${{ needs.run-tests.outputs.active-branch-coverage }}
+      DEFAULT_BRANCH_COVERAGE: ${{ needs.run-tests.outputs.default-branch-coverage }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Update coverage
+        run: |
+          echo "{ \"coverage\": $ACTIVE_BRANCH_COVERAGE }" > coverage.json
+
+      - name: Checkout/create branch, commit, and force push
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git checkout -b update-coverage
+          git add coverage.json
+          git commit -m "chore: Update coverage.json"
+          git push -f origin update-coverage
```

Comment:
Nit: Maybe better to use an alternative branch name that is less likely to be chosen by a human? As this might overwrite someone else's work.

Something like `metamaskbot/update-coverage` or `automated/update-coverage` might work.